### ISSUE TITLE
False Positive: E1101:Instance of 'ForeignKey' has no 'title' member

### ISSUE THREAD

**Issue by VictorGob** [landscapeio/pylint-django#106]
Pylint_django show a false warning in the last line. The object "Book" has a tittle but pylint shows:

> E1101:Instance of 'ForeignKey' has no 'title' member

Coding with VSCode 1.18.1

```
import uuid  # Required for unique book instances
from django.db import models
from django.urls import reverse

# Used to generate URLs by reversing the URL patterns


class Book(models.Model):
    """
    Model representing a book (but not a specific copy of a book).
    """
    title = models.CharField(max_length=200)
    author = models.ForeignKey('Author', on_delete=models.SET_NULL, null=True)
    # Foreign Key used because book can only have one author, but authors can have multiple books
    # Author as a string rather than object because it hasn't been declared yet in the file.
    summary = models.TextField(
        max_length=1000, help_text="Enter a brief description of the book")
    isbn = models.CharField(
        'ISBN', max_length=13, help_text='13 Character <a href="https://www.isbn-international.org/content/what-isbn">ISBN number</a>')
    genre = models.ManyToManyField(
        Genre, help_text="Select a genre for this book")
    # ManyToManyField used because genre can contain many books. Books can cover many genres.
    # Genre class has already been defined so we can specify the object above.

    def __str__(self):
        """
        String for representing the Model object.
        """
        return self.title

    def get_absolute_url(self):
        """
        Returns the url to access a particular book instance.
        """
        return reverse('book-detail', args=[str(self.id)])

    def display_genre(self):
        """
        Creates a string for the Genre. This is required to display genre in Admin.
        """
        return ', '.join([genre.name for genre in self.genre.all()[:3]])
    display_genre.short_description = 'Genre'

class BookInstance(models.Model):
    """
    Model representing a specific copy of a book (i.e. that can be borrowed from the library).
    """
    id = models.UUIDField(primary_key=True, default=uuid.uuid4,
                          help_text="Unique ID for this particular book across whole library")
    book = models.ForeignKey('Book', on_delete=models.SET_NULL, null=True)
    imprint = models.CharField(max_length=200)
    due_back = models.DateField(null=True, blank=True)

    LOAN_STATUS = (
        ('m', 'Maintenance'),
        ('o', 'On loan'),
        ('a', 'Available'),
        ('r', 'Reserved'),
    )

    status = models.CharField(max_length=1, choices=LOAN_STATUS,
                              blank=True, default='m', help_text='Book availability')

    class Meta:
        ordering = ["due_back"]

    def __str__(self):
        """
        String for representing the Model object
        """
        return '%s (%s)' % (self.id, self.book.title) #ERROR here, on the second "self"
```

### pylint --version output
pylint 1.7.2,
astroid 1.5.3
Python 3.6.2 (v3.6.2:5fd33b5, Jul  8 2017, 04:57:36) [MSC v.1900 64 bit (AMD64)]

**Comment by atodorov** [comment]
Duplicate of #110. The error is raised when the FK model is specified as a string, rather than class name identifier.

**Comment by pseudorandomcoder** [comment]
Fixed the false errors by changing the following line:

        # old code: return '{0} ({1})'.format(self.id, self.book.title)
        return '{0} ({1})'.format(self.id, self.book.__str__())

Also passed the model instead of a string name of the model, but it's probably not needed:

   # old code: book = models.ForeignKey('Book', on_delete=models.SET_NULL, null=True) 
    book = models.ForeignKey(Book, on_delete=models.SET_NULL, null=True)  #

**Comment by atodorov** [comment]
@pseudorandomcoder what do you mean with your latest comment? Is this happening again? If it is please open a new issue for it and copy the last comment.

**Comment by pseudorandomcoder** [comment]
I could be doing something wrong. This youtube video shows what I see and
how I worked around it. Hope this helps!

https://www.youtube.com/watch?v=n1E0qf_Dbvk&t=35s


- Jason

On Wed, May 16, 2018 at 2:52 AM, Alexander Todorov <notifications@github.com

**Comment by hack-protect** [comment]
> return '{0} ({1})'.format(self.id, self.book.__str__())

You are genius! WOW! Now i can move furthet in-depth learning of Django framework.Thank you!

**Comment by iamgilwell** [comment]
> Fixed the false errors by changing the following line:
> 
> ```
>     # old code: return '{0} ({1})'.format(self.id, self.book.title)
>     return '{0} ({1})'.format(self.id, self.book.__str__())
> ```
> 
> Also passed the model instead of a string name of the model, but it's probably not needed:
> 
> ```
> # old code: book = models.ForeignKey('Book', on_delete=models.SET_NULL, null=True) 
> book = models.ForeignKey(Book, on_delete=models.SET_NULL, null=True) 
> ```

Thanks alot, I had the same issue, solved it by         `return f'{self.category.__str__()}'`
